### PR TITLE
Improve REST API error reporting

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Log unexpected errors instead of just returning a 500 result and suppressing the exception.
Handle NodeSyncingException and report as 503 status.

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java [hunk 1]
```diff
@@ -161,19 +166,32 @@ private String readResource(final String fileName, Charset charset) throws IOExc
...
 
   private void addExceptionHandlers() {
+    app.exception(ChainDataUnavailableException.class, (e, ctx) -> ctx.status(SC_NO_CONTENT));
     app.exception(
-        ChainDataUnavailableException.class,
+        NodeSyncingException.class,
         (e, ctx) -> {
-          ctx.status(SC_NO_CONTENT);
+          ctx.status(SC_SERVICE_UNAVAILABLE);
+          setErrorBody(ctx, () -> BadRequest.serviceUnavailable(jsonProvider));
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java:174]: this breaks a lot of the old api's if we care about that?

Comment (ajsutton) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java:174]: The diff makes it look a bit confusing.  `ChainDataUnavailableException` still returns `SC_NO_CONTENT` as always.  `NodeSyncingException` previously return 500 internal server error but now returns `SC_SERVICE_UNAVAILABLE`.

Comment (rolfyone) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java:174]: ah ok

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**